### PR TITLE
Simplify the selfhosted AU guide

### PR DESCRIPTION
**Pull request by hugoShaka**
Since https://github.com/gravitational/teleport/pull/35150 and https://github.com/gravitational/teleport/pull/38937, automatic updates for self-hosted users became much easier 🎉 

Now the proxy service acts as a version server and advertises its version by default. No setup is required in the happy path.

I also removed the check "make sure you are running the enterprise Kube agent" because that's not a requirement. We have no technical or intentional reason to limit AUs for OSS users; we just YOLO-ed everything in prod and never created the teleport-updater package because the package publishing pipeline was too slow, we did not want to add yet another package.

AUs will probably come for every self-hosted setup when we'll do the updater v2.

### CODE REVIEW

File: docs/pages/includes/config-reference/proxy-service.yaml [hunk 1]
```diff
@@ -161,3 +161,18 @@ proxy_service:
         # An absolute path to the file containing the API token for OpenAI.
         # This is required for Assist functionality.
         api_token_path: /etc/teleport/openai_key
+
+    # Configuration for the built-in version server for agent automatic updates.
+    # If you leave this configuration unset, a default channel is created and
+    # serves the teleport version run by the proxy.
```
Comment (ptgott) [inline docs/pages/includes/config-reference/proxy-service.yaml:167]:
```suggestion
    # serves the Teleport version run by the Proxy Service.
```
Clarifying that this is the Teleport Proxy Service, not another proxy the user might have running in their infrastructure.

File: docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx [hunk 2]
```diff
@@ -23,216 +23,68 @@ present in their package repositories. As Teleport 14 won't be published to
...
-and reachable by all agents. You must have either:
-  - A public Amazon S3 or Google Cloud Storage bucket.
-  - A web server accessible from all agents with valid TLS certificates.
-
-If you do not have an S3 or GCS bucket for your version server, this guide
-provides an example of how to create one using Terraform. You can [install
-Terraform](https://developer.hashicorp.com/terraform/downloads) to spin up a
-demo version server along with this guide so you can get started with automatic
-agent upgrades. Terraform is not required to set up a version server. 
+<Admonition type="note">
```
Comment (ptgott) [inline docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx:36]: I think this admonition would make more sense in the changelog. Users reading this guide won't see instructions to set up a public web server. We could also provide a brief architectural overview in the introduction of this page that indicates that the Proxy Service hosts the version server. This way, we don't risk adding noise for users with an admonition.

File: docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx [hunk 3]
```diff
@@ -23,216 +23,68 @@ present in their package repositories. As Teleport 14 won't be published to
...
+   An earlier version of this guide required you to setup a public web server to
+   host the Teleport version, this is not required anymore as the Teleport proxy
+   service can now act as a version server.
+</Admonition>
 
 - (!docs/pages/includes/tctl.mdx!)
 
-## Step 1/5. Create release channel files
+## Step 1/4. Check the version served by the proxy
```
Comment (ptgott) [inline docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx:44]: I would avoid using "the proxy" if we're referring to the Teleport Proxy Service. This (a) keeps terms consistent in the docs and (b) distinguishes the Teleport Proxy Service from other proxy servers running in the user's infrastructure.

File: docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx [hunk 4]
```diff
@@ -23,216 +23,68 @@ present in their package repositories. As Teleport 14 won't be published to
...
 
-A **release channel** contains two pieces of information: the targeted version
-and if the upgrade is critical. Updaters subscribe to a release channel and will
-upgrade to the provided version during a **maintenance window** (which we will
-configure later in this guide). If the upgrade is critical, updaters will ignore
-the maintenance schedule and upgrade as soon as possible.
+The Teleport Proxy serviceexposes by default a version channel called `default`
```
Comment (ptgott) [inline docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx:46]:
```suggestion
The Teleport Proxy service exposes by default a version channel called `default`
```

File: docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx [hunk 5]
```diff
@@ -23,216 +23,68 @@ present in their package repositories. As Teleport 14 won't be published to
...
+and serving the same version the proxy is running.
 
-The version server is a static file server that responds to the following queries:
+If your proxy public address is <Var value="teleport.example.com:443" />, you can
+query the version server with the following command:\
```
Comment (ptgott) [inline docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx:50]:
```suggestion
query the version server with the following command:
```

File: docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx [hunk 6]
```diff
@@ -23,216 +23,68 @@ present in their package repositories. As Teleport 14 won't be published to
...
-## Step 1/5. Create release channel files
+## Step 1/4. Check the version served by the proxy
 
-A **release channel** contains two pieces of information: the targeted version
-and if the upgrade is critical. Updaters subscribe to a release channel and will
-upgrade to the provided version during a **maintenance window** (which we will
-configure later in this guide). If the upgrade is critical, updaters will ignore
-the maintenance schedule and upgrade as soon as possible.
+The Teleport Proxy serviceexposes by default a version channel called `default`
+and serving the same version the proxy is running.
```
Comment (ptgott) [inline docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx:47]:
```suggestion
that serves the same version the Proxy Service is running.
```

File: docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx [hunk 7]
```diff
@@ -23,216 +23,68 @@ present in their package repositories. As Teleport 14 won't be published to
...
-   </Tabs>
-
-1. Make your cloud provider credentials available to Terraform. The method to
-   use depends on your organization, e.g., pasting environment variables into
-   your terminal.
+<Details title="Changing which version is served by the version server">
+
+By default, the version server has a single `default` channel, serving the proxy
+version. If you want to override the default version or add other channels you
+can pass custom proxy configuration:
```
Comment (ptgott) [inline docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx:61]:
```suggestion
By default, the version server has a single `default` channel, serving the
version of the Teleport Proxy Service. If you want to override the default version or add other channels you
can use the `automatic_upgrades_channels` field in the Proxy Service configuration file:
```

File: docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx [hunk 8]
```diff
@@ -23,216 +23,68 @@ present in their package repositories. As Teleport 14 won't be published to
...
+
+```yaml
+proxy_service:
+  enabled: "yes"
+  automatic_upgrades_channels:
+    # Override the default version channel
+    default:
+      static_version: v14.2.1
+    # Define a new version channel with a static version
+    additional/channel/static:
```
Comment (ptgott) [inline docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx:71]: So any `/`-separated string key defines a route on the version server? Which path would the user expect to query here (is there a prefix for the path?)?

Comment (hugoShaka) [reply docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx:71]: I added a URL example and removed the slashes as it was confusing. We support slashes in the channel name, but they're not mandatory.

File: docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx (hunk 2 above)
Comment (hugoShaka) [reply docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx:36]: I removed the note, we already have a changelog entry for this

**Review by r0mant — APPROVED** [review]

File: docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx [hunk 9]
```diff
@@ -23,216 +23,62 @@ present in their package repositories. As Teleport 14 won't be published to
...
+    # https://<Var value="teleport.example.com:443" />/v1/webapi/automaticupgrades/channel/m-static-channel/version
+    my-static-channel:
+      static_version: v14.2.0
+    # Define a new version channel forwarding requests to an upstream version server
+    my-remote-channel:
+      forward_url: https://updates.releases.teleport.dev/v1/stable/cloud
+```
 
-1. Apply the configuration:
+You must ensure all proxies share the same `automatic_upgrades_channels`
```
Comment (ptgott) [inline docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx:72]:
```suggestion
You must ensure all Proxy Service instances share the same `automatic_upgrades_channels`
```

File: docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx [hunk 10]
```diff
@@ -23,216 +23,62 @@ present in their package repositories. As Teleport 14 won't be published to
...
+configuration. If proxies are configured differently, you will experience agents
```
Comment (ptgott) [inline docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx:73]:
```suggestion
configuration. If Proxy Service instances are configured differently, you will experience agents
```

File: docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx [hunk 11]
```diff
@@ -23,216 +23,62 @@ present in their package repositories. As Teleport 14 won't be published to
...
+flickering between versions as the version served is not consistent across
+proxies.
```
Comment (ptgott) [inline docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx:75]:
```suggestion
instances.
```

File: docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx [hunk 12]
```diff
@@ -23,216 +23,62 @@ present in their package repositories. As Teleport 14 won't be published to
...
-A **release channel** contains two pieces of information: the targeted version
-and if the upgrade is critical. Updaters subscribe to a release channel and will
-upgrade to the provided version during a **maintenance window** (which we will
-configure later in this guide). If the upgrade is critical, updaters will ignore
-the maintenance schedule and upgrade as soon as possible.
+The Teleport Proxy service exposes by default a version channel called `default`
+that serves the same version the Proxy Service is running.
 
-The version server is a static file server that responds to the following queries:
+If your proxy public address is <Var value="teleport.example.com:443" />, you can
```
Comment (ptgott) [inline docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx:41]:
```suggestion
If your proxy public address is <Var name="teleport.example.com:443" />, you can
```

File: docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx [hunk 13]
```diff
@@ -23,216 +23,62 @@ present in their package repositories. As Teleport 14 won't be published to
...
+query the version server with the following command:
 
 ```code
-$ curl https://<hosting-domain-and-path>/current/version
+$ curl "https://<Var value="teleport.example.com:443" />/v1/webapi/automaticupgrades/channel/default/version"
```
Comment (ptgott) [inline docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx:45]:
```suggestion
$ curl "https://<Var name="teleport.example.com:443" />/v1/webapi/automaticupgrades/channel/default/version"
```

File: docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx [hunk 14]
```diff
@@ -23,216 +23,62 @@ present in their package repositories. As Teleport 14 won't be published to
...
+By default, the version server has a single `default` channel, serving the
+version of the Teleport Proxy Service. If you want to override the default version or add other channels you
+can use the `automatic_upgrades_channels` field in the Proxy Service configuration file:
+
+```yaml
+proxy_service:
+  enabled: "yes"
+  automatic_upgrades_channels:
+    # Override the default version channel reachable at
+    # https://<Var value="teleport.example.com:443" />/v1/webapi/automaticupgrades/channel/default/version
```
Comment (ptgott) [inline docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx:60]:
```suggestion
    # https://<Var name="teleport.example.com:443" />/v1/webapi/automaticupgrades/channel/default/version
```

File: docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx [hunk 15]
```diff
@@ -23,216 +23,62 @@ present in their package repositories. As Teleport 14 won't be published to
...
+    default:
+      static_version: v14.2.1
+    # Define a new version channel with a static version reachable at
+    # https://<Var value="teleport.example.com:443" />/v1/webapi/automaticupgrades/channel/m-static-channel/version
```
Comment (ptgott) [inline docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx:64]:
```suggestion
    # https://<Var name="teleport.example.com:443" />/v1/webapi/automaticupgrades/channel/m-static-channel/version
```

File: docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx [hunk 16]
```diff
@@ -421,10 +254,11 @@ Follow these instructions on each of your Teleport agents.
    to the right release channel:
    
    ```code
-   $ echo <Var name="version-server-url/path" />/<Var name="release-channel" /> | sudo tee /etc/teleport-upgrade.d/endpoint
+   $ echo "<Var value="teleport.example.com:443" />/v1/webapi/automaticupgrades/channel/default" | sudo tee /etc/teleport-upgrade.d/endpoint
```
Comment (ptgott) [inline docs/pages/upgrading/self-hosted-automatic-agent-updates.mdx:257]:
```suggestion
   $ echo "<Var name="teleport.example.com:443" />/v1/webapi/automaticupgrades/channel/default" | sudo tee /etc/teleport-upgrade.d/endpoint
```

**Review by ptgott — APPROVED** [review]
Approved with minor product messaging notes and fixes to `Var` component instances.

**Pull request merged**